exasol_advanced_analytics_framework/testing/mock_query_handler_runner.py

```diff
@@ -0,0 +1,102 @@
+from typing import Callable, TypeVar, Generic, Tuple
+
+from exasol_data_science_utils_python.udf_utils.sql_executor import SQLExecutor
+
+from exasol_advanced_analytics_framework.query_handler.context.scope_query_handler_context import \
+    ScopeQueryHandlerContext
+from exasol_advanced_analytics_framework.query_handler.context.top_level_query_handler_context import \
+    TopLevelQueryHandlerContext
+from exasol_advanced_analytics_framework.query_handler.query.select_query import SelectQueryWithColumnDefinition
+from exasol_advanced_analytics_framework.query_handler.query_handler import QueryHandler
+from exasol_advanced_analytics_framework.query_handler.result import Continue, Finish, Result
+from exasol_advanced_analytics_framework.query_result.mock_query_result import MockQueryResult
+from exasol_advanced_analytics_framework.udf_framework.query_handler_runner_state import QueryHandlerRunnerState
+
+ResultType = TypeVar("ResultType")
+ParameterType = TypeVar("ParameterType")
+
+
+class MockQueryHandlerRunner(Generic[ParameterType, ResultType]):
+
+    def __init__(self,
+                 sql_executor: SQLExecutor,
+                 top_level_query_handler_context: TopLevelQueryHandlerContext,
+                 parameter: ParameterType,
+                 query_handler_factory: Callable[
+                     [ParameterType, ScopeQueryHandlerContext],
+                     QueryHandler[ParameterType, ResultType]]):
+        self._sql_executor = sql_executor
+        query_handler = query_handler_factory(parameter, top_level_query_handler_context)
+        self._state = QueryHandlerRunnerState(
+            top_level_query_handler_context=top_level_query_handler_context,
+            query_handler=query_handler
+        )
+
+    def run(self) -> ResultType:
+        try:
+            result = self._state.query_handler.start()
+            if isinstance(result, Finish):
+                self.handle_finish()
+                return result.result
+            else:
+                while isinstance(result, Continue):
+                    result = self.handle_continue(result)
+                if isinstance(result, Finish):
+                    self.handle_finish()
+                    return result.result
+                else:
+                    raise RuntimeError("Unknown Result")
```

Comment:
Does it really need the outer if/else branch?
I mean:

```suggestion
            while isinstance(result, Continue):
                result = self.handle_continue(result)
            if isinstance(result, Finish):
                self.handle_finish()
                return result.result
            else:
                raise RuntimeError("Unknown Result")
```

Reply:
Nie, that you wrote, you are probably right. This a leftover from the construction. I start often case by case